### PR TITLE
Update Gemfile and readme

### PR DESCRIPTION
**Pull request by ananyo2012**
This PR contains the gem updates and readme additions along with the new devise methods that had to be made after devise update. There are also some updates to pass the lint checks and failing tests. After updating the `bootstrap-sass` gem the bootstrap glyphicons that were broken previously now show correctly.

@sonalkr132 Please review this PR.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,4 +8,4 @@ public/assets
 !public/assets/images/
 !public/assets/small/
 log/
-
+Gemfile.lock
```
Comment (sonalkr132) [inline .gitignore]: `Gemfile.lock` should be tracked with git. Please remove this.

Comment (ananyo2012) [reply .gitignore]: Yeah I just updated it.

**Comment by ananyo2012** [comment]
@sonalkr132 Ready for merge now!

**Comment by sonalkr132** [comment]
Please add `secrets` variables to production.rb if you want to remove production.rb.example.

File: db/migrate/20140529165009_create_activities.rb [hunk 2]
```diff
@@ -16,6 +16,7 @@ def self.up
     add_index :activities, [:owner_id, :owner_type]
     add_index :activities, [:recipient_id, :recipient_type]
   end
+
```
Comment (sonalkr132) [inline db/migrate/20140529165009_create_activities.rb:19]: Avoid making these kind of changes.

File: spec/factories/user.rb [hunk 3]
```diff
@@ -13,7 +13,7 @@
     password 'secret12345'
 
     after(:create) do |user|
-      user.confirm!
+      user.confirm
```
Comment (sonalkr132) [inline spec/factories/user.rb:16]: Why was this needed?

**Comment by ananyo2012** [comment]
Well there are no secret tokens currently present in production.rb and production details won't be added with feature branches isn't it?

File: spec/factories/user.rb (hunk 3 above)
Comment (ananyo2012) [reply spec/factories/user.rb:16]: With the new devise version this format is deprecated. This is the new way as per devise documentation

File: db/migrate/20140529165009_create_activities.rb (hunk 2 above)
Comment (ananyo2012) [reply db/migrate/20140529165009_create_activities.rb:19]: Did this to pass the lint checks, else they were failing.

**Comment by sonalkr132** [comment]
I was talking about [these](https://github.com/ananyo2012/cat-forum/blob/860bbb9f501960142c9589a23cfb48f59a9dfee2/config/environments/production.rb#L84-L96).
Setting like those are user dependent and should not exist in version control(even if is a branch just for deployment).

**Comment by ananyo2012** [comment]
ok then perhaps we should continue using production.rb.example?

**Comment by sonalkr132** [comment]
I like the idea that all the configurable settings of the app exist in one place: `secrets.yaml` .

**Comment by ananyo2012** [comment]
@sonalkr132 Made the revisions. Have a look.

File: Gemfile [hunk 4]
```diff
@@ -1,46 +1,64 @@
...
 gem 'omniauth-facebook'
 
-gem "haml"
-
-gem 'pg'
+# Templating engine for HTML
+gem 'haml'
 # Use SCSS for stylesheets
-gem 'sass-rails', '~> 4.0.3'
+gem 'sass-rails', '>= 3.2'
```
Comment (sonalkr132) [inline Gemfile:44]: Why downgrade it?

File: README.markdown [hunk 5]
```diff
@@ -1,26 +1,59 @@
...
+* For first time installation run `rake db:setup` to setup the database
 * Start the server with `rails s`
-* Go to `localhost:3000` in your broswer.
+* Go to `localhost:3000` in your broswer
+
+## Using database migrations
+What `rake db:setup` does is that it creates the development and test database using `rake db:create`, populates the database using the schema.rb file and then seeds the data using `rake db:seed`. When a new migration is introduced you don't need to run `rake db:setup` again, instead run `rake db:migrate` to run the migrations and update the database schema.
+
+# Testing
+Run tests using the command `rspec` or `bundle exec rspec`. We are using Rspec as your Rails test suite. Writing good tests ensure that all new features introduced don't break previous ones. So consider writing tests  whenever you work on new features.
```
Comment (sonalkr132) [inline README.markdown]: Rspec is not a test suite.

File: README.markdown [hunk 6]
```diff
@@ -1,26 +1,59 @@
...
 
+# Ruby code analyzing using Rubocop
+We are using Rubocop for code analyzing and lint-checking for Ruby code. It follows the [Ruby Style Guide](https://github.com/bbatsov/ruby-style-guide). Cleaner code is always easier to read and handle. So we are using Rubocop to encourage writing of cleaner code. Please run `rubocop` before submitting any code to check there are any lint errors and correct them according to the [Ruby Style Guide](https://github.com/bbatsov/ruby-style-guide). Remember the travis-ci build will fail in case there are any lint errors.
```
Comment (sonalkr132) [inline README.markdown]: for _code_ analyzing ...ing for Ruby code :x:

File: Gemfile (hunk 4 above)
Comment (ananyo2012) [reply Gemfile:44]: Not actually downgraded. Version is `>=3.2` So it will use the latest version above 3.2. Actually I have set the parameters according to the `bootstrap-sass` [documentation](https://github.com/twbs/bootstrap-sass#a-ruby-on-rails).

File: config/database.yml.example [hunk 7]
```diff
@@ -17,7 +17,6 @@ default: &default
   port: 5432
   username: postgres
   password: 1234
-  database: mydb
```
Comment (sonalkr132) [inline config/database.yml.example:20]: Why remove database?

Comment (ananyo2012) [reply config/database.yml.example:20]: Database is added separately for each environment.

File: README.markdown (hunk 6 above)
Comment (ananyo2012) [reply README.markdown]: Ah I thought it would be better to mention this so that people would run this before pushing any code. There can be failures in the build due to the link checks.

File: README.markdown (hunk 5 above)
Comment (ananyo2012) [reply README.markdown]: Uh it is TDD framework. So how does it sound if I put its as `We are using Rspec as our testing framework` ?

File: README.markdown (hunk 6 above)
Comment (sonalkr132) [reply README.markdown]: You have used _code_ twice. I am asking to correct that.

Comment (ananyo2012) [reply README.markdown]: Uh ok just updating it.

**Comment by ananyo2012** [comment]
@sonalkr132 Updated!

**Comment by sonalkr132** [comment]
:+1:
